api/query/search.go

```diff
@@ -99,68 +103,97 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
+		r2 := r.Clone(r.Context())
+		r2.Method = "GET"
+		r2.URL.RawQuery = q.Encode()
+		summariesValid, err := sh.validateSummaryVersions(r2.URL.Query())
 		if err != nil {
-			logger.Errorf("Failed to create request to POST %s: %v", fwdURL.String(), err)
-			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
-			return
+			logger.Debugf("Error checking version files: %v", err)
 		}
-		req.Header.Add("Content-Type", "application/json")
+		isSimpleQ = isSimpleQ && summariesValid
+	}
 
-		resp, err := client.Do(req)
+	// Use searchcache for a complex query or if old summary files exist.
+	if !isSimpleQ {
+		resp, err := sh.useSearchcache(w, r, data, logger)
 		if err != nil {
-			logger.Errorf("Error connecting to search API cache: %v", err)
 			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
-			return
-		}
-
-		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-			msg := fmt.Sprintf("Error from request: POST %s: STATUS %d", fwdURL.String(), resp.StatusCode)
-			errBody, err2 := ioutil.ReadAll(resp.Body)
-			if err2 == nil {
-				msg = fmt.Sprintf("%s: %s", msg, string(errBody))
-				resp.Body = ioutil.NopCloser(bytes.NewBuffer(errBody))
+		} else {
+			defer resp.Body.Close()
+			w.WriteHeader(resp.StatusCode)
+			_, err = io.Copy(w, resp.Body)
+			if err != nil {
+				logger.Errorf("Error forwarding response payload from search cache: %v", err)
 			}
-			logger.Errorf(msg)
-		}
-
-		defer resp.Body.Close()
-		w.WriteHeader(resp.StatusCode)
-		_, err = io.Copy(w, resp.Body)
-		if err != nil {
-			logger.Errorf("Error forwarding response payload from search cache: %v", err)
 		}
 		return
 	}
+	sh.handleSimpleQuery(w, r, simpleQ, rq)
+}
+
+func (sh structuredSearchHandler) handleSimpleQuery(w http.ResponseWriter,
+	r *http.Request, simpleQ TestNamePattern, rq RunQuery) {
+	// Assemble list of run IDs for later use.
+	runIDStrs := make([]string, 0, len(rq.RunIDs))
+	for _, id := range rq.RunIDs {
+		runIDStrs = append(runIDStrs, strconv.FormatInt(id, 10))
+	}
+	runIDsStr := strings.Join(runIDStrs, ",")
 
 	// Structured query is equivalent to unstructured query.
 	// Create an unstructured query request and delegate to unstructured query
 	// handler.
-	r2 := *r
+	r2 := r.Clone(r.Context())
 	r2url := *r.URL
 	r2.URL = &r2url
 	r2.Method = "GET"
-	runIDStrs := make([]string, 0, len(rq.RunIDs))
-	for _, id := range rq.RunIDs {
-		runIDStrs = append(runIDStrs, strconv.FormatInt(id, 10))
+	r2.URL.RawQuery = fmt.Sprintf("run_ids=%s&q=%s",
+		url.QueryEscape(runIDsStr), url.QueryEscape(simpleQ.Pattern))
+
+	unstructuredSearchHandler{queryHandler: sh.queryHandler}.ServeHTTP(w, r2)
+}
+
+func (sh structuredSearchHandler) useSearchcache(w http.ResponseWriter, r *http.Request, data []byte, logger shared.Logger) (*http.Response, error) {
+	hostname := sh.api.GetServiceHostname("searchcache")
+	// TODO: This will not work when hostname is localhost (http scheme needed).
```

Comment:
I know this TODO is already here but could you log a github issue and add the github issue link/number here please?

Reply:
This is a comment I moved that existed previously, so I'm not sure of the context here, but I've created [an issue](https://github.com/web-platform-tests/wpt.fyi/issues/2941) and I'll link it in the comment.